packages/byods/src/BYODS.ts

```diff
@@ -1,46 +1,48 @@
-/* eslint-disable no-console */
-import fetch from 'node-fetch';
+import {jwksCache, createRemoteJWKSet} from 'jose';
+import BaseClient from './BaseClient';
+import {USER_AGENT, JWKS_URL, DEFAULT_BASE_URL} from './constants';
+import {SDKConfig, ServiceAppAuthorizations} from './types';
+import TokenManager from './TokenManager';
 
-interface SDKConfig {
-  clientId: string;
-  clientSecret: string;
-  accessToken: string;
-  refreshToken: string;
-  expiresAt: Date;
-}
+export default class BYODS {
+  private serviceAppId: string;
+  private headers: Record<string, string> = {
+    'User-Agent': USER_AGENT,
+  };
 
-class BYODS {
-  private clientId: string;
-  private clientSecret: string;
-  private tokenHost: string;
-  private accessToken: string;
-  private refreshToken: string;
-  private expiresAt: Date;
+  private jwksCache: Record<string, never> = {};
+  private jwks: any;
 
-  constructor({clientId, clientSecret, accessToken, refreshToken, expiresAt}: SDKConfig) {
-    this.clientId = clientId;
-    this.clientSecret = clientSecret;
-    this.accessToken = accessToken;
-    this.refreshToken = refreshToken;
-    this.expiresAt = expiresAt;
-    this.tokenHost = 'https://webexapis.com/v1/access_token';
-  }
+  private config: SDKConfig;
+  private serviceAppAuthorizations: ServiceAppAuthorizations = {};
+
+  public tokenManager: TokenManager;
 
-  public async makeAuthenticatedRequest(endpoint: string): Promise<any> {
-    if (new Date() >= new Date(this.expiresAt as Date)) {
-      throw new Error('Token has expired');
-    }
+  constructor({clientId, clientSecret, baseUrl = DEFAULT_BASE_URL}: SDKConfig) {
+    this.config = {clientId, clientSecret, baseUrl};
+    this.serviceAppId = Buffer.from(`ciscospark://us/APPLICATION/${clientId}`).toString('base64');
+    this.tokenManager = new TokenManager(clientId, clientSecret);
 
-    // Use this.token.access_token to make authenticated requests
-    const response = await fetch(endpoint, {
-      headers: {
-        Authorization: `Bearer ${this.accessToken}`,
-      },
+    // Create a remote JWK Set
+    this.jwks = createRemoteJWKSet(new URL(JWKS_URL), {
+      [jwksCache]: this.jwksCache,
+      cacheMaxAge: 600000, // 10 minutes
+      cooldownDuration: 30000, // 30 seconds
     });
+  }
 
-    return response.json();
+  /**
+   * Retrieves a client instance for a specific organization.
+   *
+   * @param {string} orgId - The unique identifier of the organization.
+   * @returns {BaseClient} A new instance of BaseClient configured for the specified organization.
+   */
+  public getClientForOrg(orgId: string): BaseClient {
```

Comment:
Should we throw an error if `orgId` is not passed?

Reply:
Updated the code